### ISSUE TITLE
[Bug][config-ui] Clicking `save and run now` doesn't refresh the blueprint page and show the in-progress run

### ISSUE THREAD

**Issue by Startrekzky** [apache/incubator-devlake#3237]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

After I finished configuring a blueprint and clicked the `save and run now`, the page looked like this:

![2faecece-2430-46db-b5c1-c917a0068163](https://user-images.githubusercontent.com/14050754/192812116-b4538b39-5522-48c1-9b4f-6ecc118d49d1.jpeg)

I thought `save and run now` was broken, so I clicked 3 times again. Then I refreshed the page, and it showed a current run and 3 historical runs:

![image](https://user-images.githubusercontent.com/14050754/192812718-731f1f48-8941-4269-846b-8e6b39d230dc.png)

I'm not sure if these 4 pipelines executed concurrently will slow down the collection speed, if so, this bug can be very problematic.

### What you expected to happen

This page auto-refresh and show the current run.

### How to reproduce

Visit here or your own config-ui and create a blueprint: https://ui-my-svr02.demo.devlake.io/

### Version

main

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by likyh** [comment]
@e2corporation Are you focused on this issue? If not, would like to allow me to fix this issue and refactor some code on this page? You know, this page has several bugs caused by code. Such as setInteval will not finish when pipeline is success; send the same request at the same times; send no use request when edit blueprint.
![image](https://user-images.githubusercontent.com/3294100/192822162-b9ee1773-bfa5-4e05-b340-e77f14ee73ac.png)
![image](https://user-images.githubusercontent.com/3294100/192822545-392eb2c8-0b64-4a38-9de1-a73520641efd.png)
![image](https://user-images.githubusercontent.com/3294100/192822876-c465c3c2-3bbb-4d77-a093-100b588c7e89.png)

**Comment by e2corporation** [comment]
@likyh That last screenshot where "gitlab" is in the connections is most likely due to all connections being fetched, this is not a bug.

**Comment by e2corporation** [comment]
@likyh I don't think there's anything refactoring needed at this point, what exact bugs do you mean by "several bugs" ?

**Comment by likyh** [comment]
> @likyh That last screenshot where "gitlab" is in the connections is most likely due to all connections being fetched, this is not a bug.

@e2corporation Snapshot 2 show that there is only one plugin. It's because the default value and the page just use connection from /connections instead of /connection/:id

**Comment by e2corporation** [comment]
In screenshot 2 all the JIRA proxy endpoints are being hit because they were requested, that seems normal. Fields ,Boards and Issue Types are being fetched for connection ID 1, the XHR log looks fine there are no missing "connection id" strings

**Comment by e2corporation** [comment]
@likyh If you'd like to do some tasteful refactoring and feel that there are valid issues to address on Blueprint Detail then proceed, keep in mind we are frozen for a release so too much refactoring may introduce new bugs.

**Comment by likyh** [comment]
> @likyh I don't think there's anything refactoring needed at this point, what exact bugs do you mean by "several bugs" ?

Let's look at the bugs one by one.

1. ~the `setInterval` will never stop. (fixed before)~

2. When open this page, the delicate request `pipelines/17` and `pipelines/17/tasks` are sent at once.
![image](https://user-images.githubusercontent.com/3294100/192825809-dfbaa0ca-3e66-4215-817b-49f6d48a471a.png)

3. the request `/api/plugins/gitlab/connection/1` is sent when try to edit jira connection. There are no gitlab connection.
![image](https://user-images.githubusercontent.com/3294100/192824880-efd91585-f93f-4f79-a115-405262200276.png)

**Comment by e2corporation** [comment]
> > @likyh I don't think there's anything refactoring needed at this point, what exact bugs do you mean by "several bugs" ?
> 
> Let's look at the bugs one by one.
> 
> 1. ~the `setInterval` will never stop. (fixed before)~
> 2. When open this page, the delicate request `pipelines/17` and `pipelines/17/tasks` are sent at once.
>    ![image](https://user-images.githubusercontent.com/3294100/192825809-dfbaa0ca-3e66-4215-817b-49f6d48a471a.png)
Yes this is desired behavior, tasks are not bundled with the pipeline details, we associate the related tasks to the pipeline object after fetching related tasks.
> 3. the request `/api/plugins/gitlab/connection/1` is sent when try to edit jira connection. There are no gitlab connection.
>    ![image](https://user-images.githubusercontent.com/3294100/192824880-efd91585-f93f-4f79-a115-405262200276.png)
There isn't a gitlab connection here yes on the blueprint, but at a higher level we need to be aware of all connections available (DSM Hook builds the all connections list). So what you are seeing is the effect of All connections being fetched.

**Comment by likyh** [comment]
Another 2 bugs found in this page:
1. https://github.com/apache/incubator-devlake/pull/3287#pullrequestreview-1126231661
2. https://github.com/apache/incubator-devlake/pull/3287#issuecomment-1263105695

@e2corporation I think this page needs to be refactor.

**Comment by Startrekzky** [comment]
Verified.

**Comment by e2corporation** [comment]
> Another 2 bugs found in this page:
> 
> 1. [[Bug][config-ui] board transfomation match wrong when edit jira blueprint #3294](https://github.com/apache/incubator-devlake/issues/3294)
> 2. [[Bug][config-ui] add board but type mapping is wrong when edit jira blueprint #3295](https://github.com/apache/incubator-devlake/issues/3295)
> 
> @e2corporation I think this page needs to be refactored.

@likyh This should have been fixed with my last PR for "JIRA Modification Degraded..."
